File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.cpp
#include <string>
#include <vector>
#include <sstream>
#include <unordered_set>
#include <unordered_map>

using namespace std;

vector<int> solution(vector<string> id_list, vector<string> report, int k) {
    vector<int> answer;
    unordered_map<string, int> count;
    unordered_map<string, unordered_set<string>> reported_user;
    
    for (string& str : report) {
        stringstream ss(str);
        string user_id, reported_id;
        ss >> user_id >> reported_id;
        reported_user[reported_id].insert(user_id);
    }
    
    for (auto& [reported_id, user_id_lst] : reported_user) {
        if (user_id_lst.size() >= k) {
            for (const string& uid : user_id_lst)
                count[uid]++;
        }
    }
    
    for (string& uid : id_list) {
        answer.push_back(count[uid]);
    }
    
    return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    answer = [0]*len(id_list)
    dict = {name : 0 for name in id_list}
    
    for case in set(report):
        ing, ed = case.split()
        dict[ed] += 1
    
    for case in set(report):
        ing, ed = case.split()
        if dict[ed] >= k: answer[id_list.index(ing)] += 1
            
    
    return answer